torch/csrc/jit/passes/onnx/constant_fold.cpp

```diff
@@ -105,6 +102,91 @@ c10::optional<at::Tensor> runTorchBackendForOnnx(
       updated_val = at::narrow(updated_val, axis, start, length);
     }
     return c10::optional<at::Tensor>(updated_val);
+}
+
+c10::optional<at::Tensor> runTorchSlice_opset10(const Node* node, 
+                     std::vector<at::Tensor>& inputTensorValues) {
+  assert(inputTensorValues.size() > 2 &&  inputTensorValues.size() < 6);
+  // Checking validity of 'starts' and 'ends' input
+  assert(inputTensorValues[1].sizes().size() == 1 &&
+         inputTensorValues[2].sizes().size() == 1);             
+  if (inputTensorValues[1].sizes()[0] != inputTensorValues[2].sizes()[0] ) {
+    // Number of elements of 'starts' and 'ends' 1-D input tensors should be the same
+    return c10::nullopt;
+  }
+  // Checking 'axes' input, if available.
+  std::vector<int64_t> axes;
+  if (inputTensorValues.size() > 3) {
+    assert(inputTensorValues[3].sizes().size() == 1); 
+    if (inputTensorValues[3].sizes()[0] != inputTensorValues[1].sizes()[0] ) {
+      // Number of elements of 'axes' and 'ends' 1-D input tensors should be the same
+      return c10::nullopt;
+    }
+    auto axes_a = inputTensorValues[3].accessor<int64_t, 1>();
+    axes.reserve(inputTensorValues[3].sizes()[0]); 
+    for (size_t i = 0; i < inputTensorValues[3].sizes()[0]; ++i) {
+      axes[i] = axes_a[i];
+    }
+  }
+  else {
+    axes = std::vector<int64_t>(inputTensorValues[1].sizes()[0], 0);
+  }
+  // Checking 'steps' input, if available.
+  if (inputTensorValues.size() > 4) {
+    assert(inputTensorValues[4].sizes().size() == 1); 
+    if (inputTensorValues[4].sizes()[0] != inputTensorValues[1].sizes()[0] ) {
+      // Number of elements of 'steps' and 'ends' 1-D input tensors should be the same
+      return c10::nullopt;
+    }
+    auto steps_a = inputTensorValues[4].accessor<int64_t, 1>();
+    for (size_t i = 0; i < inputTensorValues[4].sizes()[0]; ++i) {
+      // Only steps == 1 are supported for constant-folding.
+      if (steps_a[i] != 1) {
+        return c10::nullopt;
+      }
+    }
+  }
+  auto starts_a = inputTensorValues[1].accessor<int64_t, 1>();
+  auto ends_a = inputTensorValues[2].accessor<int64_t, 1>();
+  auto updated_val = inputTensorValues[0];
+  for (size_t i = 0; i < inputTensorValues[1].sizes()[0]; ++i) {
+    // ONNX slice accepts negative starts and ends values.
+    int64_t start = starts_a[i], end = ends_a[i];
+    int64_t axis = axes[i];
```

Comment:
abstract this out to helper function for the logic of handling negative index?

Reply:
Makes sense. Fixed.